TAB2MXL/src/main/java/View/Controller.java

```diff
@@ -387,19 +388,26 @@ public void detect(String input) {
 //		System.out.println(lines.length);
 
 		// basic checks
-		if (lines[0].toUpperCase().startsWith("E")) {
+		if (lines[0].toUpperCase().startsWith("E") && (lines.length % 6==0)) {
+			//System.out.println("This is a guitar"); // testing
 			guitarButtonClicked();
 			/*
 			 * Guitar
 			 */
 
-		} else if (lines[0].toUpperCase().startsWith("C")) {
+		} else if (lines[0].toUpperCase().startsWith("C") && (lines.length % 5==0)) {
+			//System.out.println("This is a bass"); // testing
 			drumButtonClicked();
 			// Drum
-		} else if (lines[0].toUpperCase().startsWith("G")) {
+		} else if (lines[0].toUpperCase().startsWith("G") && (lines.length % 4==0)) {
+			//System.out.println("This is a bass"); // testing
 			bassButtonClicked();
 			// Bass
 		}
+		else
+		{
+			//System.out.println("Error in Instrument Detection"); //testing
```

Comment:
Throw an exception here, otherwise the program will move forward with no instrument selected which can lead to some problems